thoth/storages/graph/postgres.py

```diff
@@ -6738,6 +6740,108 @@ def get_bloat_data(self) -> List[Dict[str, Any]]:
 
         return self._process_bloat_data_results(tables=tables)
 
+    def purge_solver_documents(
+        self, *, os_name: Optional[str] = None, os_version: Optional[str] = None, python_version: Optional[str] = None
+    ) -> int:
+        """Store and purge to be deleted solver documents to Ceph"""
+        self.connect()
+        solver_store = SolverResultsStore()
+        solver_store.connect()
+
+        target_prefix = f"{solver_store.ceph.prefix.rsplit('/', maxsplit=1)[0]}/solver-purge-{datetime2datetime_str()}/"
+        target_store = CephStore(prefix=target_prefix)
+        target_store.connect()
+
+        solver_document_ids = self.get_solver_run_document_ids_all(
+            os_version=os_version, os_name=os_name, python_version=python_version
+        )
+
+        deleted_solver_documents_count = 0
+
+        for solver_document_id in solver_document_ids:
+            document = solver_store.retrieve_document(document_id=solver_document_id)
+            target_store.store_document(document, document_id=solver_document_id)
+            solver_store.ceph.delete(object_key=solver_document_id)
+            deleted_solver_documents_count += self.delete_solver_result(solver_document_id=solver_document_id)
+
+        return deleted_solver_documents_count
+
+    def purge_adviser_documents(
+        self, *, end_datetime: Optional[str] = None, adviser_version: Optional[str] = None
```

Comment:
Looking into this, should `end_datetime` be of type `datetime`? Otherwise, it is confusing. I see this also in other parts of the module - we should fix it. Let's keep this as-is for now and adjust it in a separate PR.

Reply:
You are right, I typed it as a string because I forgot that the actual argument from `purge-job` uses the `parse` method from `dateutil`, I will open a PR and correct this !